sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Scenario.Tests/tests/AzureStorageEndToEndTests.cs

```diff
@@ -160,7 +167,144 @@ public async Task AzureStorageEndToEndFast()
+            string functionId = $"{methodInfo.DeclaringType.FullName}.{methodInfo.Name}";
+            var concurrencyManager = host.Services.GetServices<ConcurrencyManager>().SingleOrDefault();
+            var concurrencyStatus = concurrencyManager.GetStatus(functionId);
+            Assert.AreEqual(1, concurrencyStatus.CurrentConcurrency);
+
+            // write a bunch of queue messages
+            int numMessages = 250;
+            string queueName = _resolver.ResolveInString(DynamicConcurrencyQueueName);
+            await WriteQueueMessages(queueName, numMessages);
+
+            // start the host
+            await host.StartAsync();
+
+            // wait for all messages to be processed
+            await TestHelpers.Await(() =>
+            {
+                return DynamicConcurrencyTestJob.InvocationCount >= numMessages;
+            });
+
+            await host.StopAsync();
+
+            // ensure we've dynamically increased concurrency
+            concurrencyStatus = concurrencyManager.GetStatus(functionId);
+            Assert.GreaterOrEqual(concurrencyStatus.CurrentConcurrency, 10);
+
+            // check a few of the concurrency logs
+            var concurrencyLogs = host.GetTestLoggerProvider().GetAllLogMessages().Where(p => p.Category == LogCategories.Concurrency).Select(p => p.FormattedMessage).ToList();
+            int concurrencyIncreaseLogCount = concurrencyLogs.Count(p => p.Contains("ProcessMessage Increasing concurrency"));
+            Assert.GreaterOrEqual(concurrencyIncreaseLogCount, 3);
+        }
+
+        [Test]
+        [Category("DynamicConcurrency")]
+        public async Task DynamicConcurrency_Blobs()
+        {
+            // Reinitialize the name resolver to avoid conflicts
+            _resolver = new RandomNameResolver();
+            DynamicConcurrencyTestJob.InvocationCount = 0;
+
+            IHost host = new HostBuilder()
+                .ConfigureDefaultTestHost<DynamicConcurrencyTestJob>(b =>
+                {
+                    b.AddAzureStorageBlobs();
+
+                    b.Services.AddOptions<ConcurrencyOptions>().Configure(options =>
+                    {
+                        options.DynamicConcurrencyEnabled = true;
+                    });
+
+                    b.Services.AddOptions<QueuesOptions>().Configure(options =>
+                    {
+                        options.MaxPollingInterval = TimeSpan.FromSeconds(1);
+                    });
+                })
+                .ConfigureServices(services =>
+                {
+                    services.AddSingleton<INameResolver>(_resolver);
+                })
+                .ConfigureLogging((context, b) =>
+                {
+                    b.SetMinimumLevel(LogLevel.Debug);
+                })
+                .Build();
+
+            string sharedListenerId = "SharedBlobQueueListener";
```

Comment:
I've added an E2E test for blobs. This shows the shared listener pattern - the fact that internally there's a virtual queue trigger function "SharedBlobQueueListener".